examples/medplum-photon-integration/src/bots/sync-patient-from-photon.ts

```diff
@@ -309,83 +307,140 @@ export function createAllergies(
   return allergies;
 }
 
-export async function createPrescriptions(
+/**
+ * Takes an array of Photon Prescription objects for a given patient and uses these to create the patient's medication
+ * history. This creates MedicationRequest resources to represent prescriptions and MedicationDispense resources to
+ * represent fills of these prescriptions. These are added to an array of Bundle entries that can be executed in a
+ * batch.
+ *
+ * @param patientReference - A reference to the Patient the MedicationRequests and MedicationDispenses are for
+ * @param medplum - Medplum Client used to get additional details from your project
+ * @param photonPrescriptions - An array of Photon Prescription objects used to create the medication history
+ * @returns An array of Bundle Entries containing the medication history of a patient
+ */
+export async function createMedicationHistoryEntries(
   patientReference: Reference<Patient>,
   medplum: MedplumClient,
   photonPrescriptions?: PhotonPrescription[]
-): Promise<MedicationRequest[] | undefined> {
+): Promise<BundleEntry<MedicationDispense | MedicationRequest>[] | undefined> {
   if (!photonPrescriptions || photonPrescriptions.length === 0) {
     return undefined;
   }
 
-  const prescriptions: MedicationRequest[] = [];
+  const entries: BundleEntry<MedicationDispense | MedicationRequest>[] = [];
   for (const photonPrescription of photonPrescriptions) {
-    if (await checkForExistingPrescription(medplum, photonPrescription)) {
```

Comment:
I'm not familiar with Photon, so bear with me here, but it seems like the new version doesn't have any way of checking for existing resources or upserting existing `MedicationRequest` resources. Is there something that prevents duplicates from being made after repeated calls of this bot?

Reply:
I think you're correct @mattlong . @ksmith94 Is there a reason you're not upserting the `MedicationRequest` resource?

Reply:
@rahul1 @mattlong Agreed this should be upserted. I think I may have removed this when I was skipping any existing patients, so there was no need to dedupe then. I'll add it back